packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -419,27 +424,26 @@ class TextSelectionOverlay {
     if (selectionControls == null)
       return Container();
 
-    // Find the horizontal midpoint, just above the selected text.
-    final List<TextSelectionPoint> endpoints = renderObject.getEndpointsForSelection(_selection);
-    final Offset midpoint = Offset(
-      (endpoints.length == 1) ?
-        endpoints[0].point.dx :
-        (endpoints[0].point.dx + endpoints[1].point.dx) / 2.0,
-      endpoints[0].point.dy - renderObject.preferredLineHeight,
-    );
-
     final Rect editingRegion = Rect.fromPoints(
       renderObject.localToGlobal(Offset.zero),
       renderObject.localToGlobal(renderObject.size.bottomRight(Offset.zero)),
     );
 
+    // Send points to a specific implementation to
```

Comment:
The specific implementation may not intend to find horizontal midpoints. Just say let the subclass know the selected positions to decide where to show the toolbar.